### PR TITLE
Add point probes to demo

### PR DESCRIPTION
**Pull request by schmitts**
Extend the probe demo example with point probes sampling the conductance of two synapses.

### CODE REVIEW

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Thanks for adding this! A C++ism and then we are good to merge.

**Review by thorstenhater — APPROVED** [review]
👍

File: example/probe-demo/probe-demo.cpp [hunk 1]
```diff
@@ -173,21 +186,29 @@ void scalar_sampler(arb::probe_metadata pm, std::size_t n, const arb::sample_rec
...
-        assert(n_cable==value_range->second-value_range->first);
-
-        for (unsigned j = 0; j<n_cable; ++j) {
-            arb::mcable where = (*cables_ptr)[j];
-            std::cout << samples[i].time << ", "
-                      << where.prox_pos << ", "
-                      << where.dist_pos << ", "
-                      << value_range->first[j] << '\n';
+        const auto& [lo, hi] = *value_range;
+        assert(n_entities==hi-low);
```
Comment (brenthuisman) [inline example/probe-demo/probe-demo.cpp:200]:
```suggestion
        assert(n_entities==hi-lo);
```

**Review by brenthuisman — CHANGES_REQUESTED** [review]
I believe this is the showstopper.

**Comment by schmitts** [comment]
> I believe this is the showstopper.

I was compiling in `release` that nulls the asserts ...

**Review by brenthuisman — APPROVED** [review]

**Pull request merged**